src/xkbcomp/rules.c

```diff
@@ -1059,15 +1065,14 @@ static bool
 read_rules_file(struct xkb_context *ctx,
                 struct matcher *matcher,
                 unsigned include_depth,
+                FILE *file,
                 const char *path)
 {
     bool ret = false;
-    FILE *file;
     char *string;
     size_t size;
     struct scanner scanner;
 
-    file = fopen(path, "r");
     if (!file)
```

Comment:
I think it should be up to the caller to guarantee this, so should remove the check here.

Reply:
ack, removed.